tardis/widgets/sdec_plot.py

```diff
@@ -502,7 +502,10 @@ def _calculate_plotting_data(
         if distance is None:
             self.lum_to_flux = 1  # so that this term will have no effect
         else:
-            self.lum_to_flux = 4.0 * np.pi * (distance.to("cm")) ** 2
+            if distance <= 0:
+                raise ValueError("distance passed must be greater than 0")
+            else:
+                self.lum_to_flux = 4.0 * np.pi * (distance.to("cm")) ** 2
 
```

Comment:
This is better, but is it possible to either have a message explaining that if D=0, this doesn't correspond to the luminosity case, and if the luminosity case is desired, then the user needs to remove the distance flag, or set to None. Alternatively, a simple workaround could be to allow D=0 to correspond to the Luminosity case, or would that confuse people? Maybe the first option is better...

Reply:
I agree, 1st option is better. Will mention that in error message.

Reply:
@jamesgillanders done! Let me know if it needs to be reworded for more clarity

Reply:
this is perfect, thanks